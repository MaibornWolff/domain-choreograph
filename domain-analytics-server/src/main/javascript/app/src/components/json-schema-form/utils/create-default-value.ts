import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';

const defaultValues: Record<Exclude<JSONSchema4TypeName, 'object'>, any> = {
  string: '',
  number: undefined,
  integer: undefined,
  boolean: false,
  array: [],
  'null': null,
  any: ''
};

export function createDefaultValue(schema: JSONSchema4): object {
  return createDefaultValueForObject(schema);
}

function createDefaultValueForObject(schema: JSONSchema4): Promise<object> {
  if (schema.type !== 'object') {
    throw new Error('Expect schema to be an object');
  }
  let defaultValue: any = {};
  if (schema.properties) {
    Object.entries(schema.properties).forEach(([key, property]) => {
      const type = property.type || 'any';
      if (type instanceof Array) {
        throw new Error('Array types are no supported');
      } else if (type === 'object') {
        defaultValue[key] = createDefaultValueForObject(property);
      } else {
        defaultValue[key] = defaultValues[type];
      }
    });
  }
  return defaultValue;
}
