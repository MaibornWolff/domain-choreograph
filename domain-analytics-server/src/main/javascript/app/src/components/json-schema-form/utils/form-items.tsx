import { styled } from '~utils/styled';

export const FormItems = styled('div')`
    display: flex;
    flex-direction: column;

    & > * {
      margin-bottom: .5rem;
    }
`;
