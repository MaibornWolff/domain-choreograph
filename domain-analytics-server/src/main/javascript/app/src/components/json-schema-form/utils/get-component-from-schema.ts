import { FieldComponent } from 'clean-forms';
import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
import { DatePickerMsField } from '~components/json-schema-form/form-fields/date-picker-ms-field';
import { NumberTextField, TextField } from '~components/json-schema-form/form-fields/text-field';
import { JsonSchemaFormFieldProps } from '~components/json-schema-form/models/json-schema-form-field-props';

export interface JsonSchemaComponentMap {
  matcher?: { type: JSONSchema4TypeName, format?: string };
  field: FieldComponent<any, JsonSchemaFormFieldProps>;
}

const defaultMapping: JsonSchemaComponentMap[] = [
  { matcher: { type: 'integer', format: 'utc-millisec' }, field: DatePickerMsField },
  { matcher: { type: 'integer' }, field: NumberTextField },
  { matcher: { type: 'number' }, field: NumberTextField },
  { field: TextField }
];

export function getComponentFromSchema(
  schema: JSONSchema4,
  mapping: JsonSchemaComponentMap[] = defaultMapping
): FieldComponent<any, JsonSchemaFormFieldProps> {
  for (const map of mapping) {
    if (map.matcher == null) {
      return map.field;
    }
    const typeIsMatching = map.matcher.type === schema.type;
    const formatIsMatching = map.matcher.format == null || map.matcher.format === schema.format;
    if (typeIsMatching && formatIsMatching) {
      return map.field;
    }
  }
  throw new Error('Couldn\'t match the schema');
}
