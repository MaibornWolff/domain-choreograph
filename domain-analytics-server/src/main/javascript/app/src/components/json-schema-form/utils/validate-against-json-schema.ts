import { ValidationFunction } from 'clean-forms';
import { JSONSchema4 } from 'json-schema';
import { Validator } from 'jsonschema';


export function validateAgainstJsonSchema(schema: JSONSchema4): ValidationFunction {
  const validator = new Validator();
  return ({ value }) => {
    const result = validator.validate(value, schema as any);
    const errors = result.errors.map(error => {
      let message = error.message;
      let property = error.property
        .replace(/^instance\.?/, '');
      let schema = error.schema as JSONSchema4;
      if (error.name === 'required') {
        message = `is required`;
        schema = schema.properties![error.argument];
        property = joinProperties(property, error.argument);
      }

      const title = schema.title || '';
      message = `${title} ${message}`;

      return [ property, message ] as [string, string];
    });
    return errors;
  }
}

function joinProperties(path1: string, path2: string): string {
  if (path1 === '') {
    return path2;
  }
  if (path2 === '') {
    return path1;
  }
  return `${path1}.${path2}`;
}
