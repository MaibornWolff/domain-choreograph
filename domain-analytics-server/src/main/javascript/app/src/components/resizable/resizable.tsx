import * as React from 'react';
import { styled } from '~utils/styled';

const Wrapper = styled('div')`
  position: relative;;
  height: 100%;
`;

const ScrollWrapper = styled('div')`
  overflow: auto;
  height: 100%;
`;

const resizeAreaWidth = 50;
const ResizeArea = styled('div')`
  position: absolute;
  width: ${resizeAreaWidth}px;
  left: -${resizeAreaWidth / 2}px;
  top: 0;
  bottom: 0;
  cursor: col-resize;
  user-select: none;
`;

export interface ResizableProps {
  className?: string;
  wrapperClassName?: string;
}

export interface ResizableState {
  isDragging: boolean;
  dragStart: number | null;
  width: number;
}

export class Resizable extends React.Component<ResizableProps, ResizableState> {
  public state: ResizableState = {
    isDragging: false,
    dragStart: null,
    width: 350,
  };

  public render() {
    return (
      <Wrapper
        style={{width: this.state.width}}
        className={this.props.wrapperClassName}
      >
        <ResizeArea
          onMouseDown={this.startDrag as any}
        />
        <ScrollWrapper className={this.props.className}>
          {this.props.children}
        </ScrollWrapper>
      </Wrapper>
    );
  }

  public componentDidMount() {
    document.addEventListener('mouseup', this.endDrag);
    document.addEventListener('mousemove', this.mouseMove);
  }

  public componentWillUnmount() {
    document.removeEventListener('mouseup', this.endDrag);
    document.removeEventListener('mousemove', this.mouseMove);
  }

  private startDrag = (event: MouseEvent) => {
    this.setState({
      isDragging: true,
      dragStart: event.clientX
    });
  }

  private endDrag = () => {
    this.setState({
      isDragging: false,
      dragStart: null
    });
  }

  private mouseMove = (event: MouseEvent) => {
    if (!this.state.isDragging || this.state.dragStart == null) {
      return;
    }
    const newPosition = event.clientX;
    const offset = this.state.dragStart - newPosition;
    this.setState(oldState => ({
      dragStart: newPosition,
      width: oldState.width + offset
    }));
  }
}
