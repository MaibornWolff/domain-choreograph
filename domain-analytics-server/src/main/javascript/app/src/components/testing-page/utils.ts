export function escapeJavaClass(javaClass: string) {
  return javaClass.replace(/\./g, '-');
}

export function reverseJavaClassEscape(javaClass: string) {
  return javaClass.replace(/-/g, '.');
}
