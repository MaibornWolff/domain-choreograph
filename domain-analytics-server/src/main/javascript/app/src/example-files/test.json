{
  "nodes": {
    "A": {
      "id": "A",
      "name": "A",
      "scope": "root"
    }
  },
  "dependencies": [],
  "scopes": {
    "root": {
      "id": "root",
      "executionContext": "Application",
      "nodes": [
        "A"
      ]
    }
  },
  "executionContexts": {
    "Application": {
      "id": "Application",
      "scopes": [
        "root"
      ]
    }
  }
}
