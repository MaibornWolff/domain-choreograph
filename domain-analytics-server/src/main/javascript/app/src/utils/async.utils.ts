export function wait(milliseconds: number): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    setTimeout(() => resolve(), milliseconds);
  });
}
