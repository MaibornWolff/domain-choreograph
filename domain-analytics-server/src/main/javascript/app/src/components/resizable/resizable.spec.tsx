import { shallow } from 'enzyme';
import * as React from 'react';
import { mockEvent } from '~utils/test.utils';
import { Resizable } from './resizable';

describe('Resizable', () => {
  it('should render', () => {
    expect(shallow(<Resizable/>)).toMatchSnapshot();
  });

  it('should set state on drag start', () => {
    const component = shallow(<Resizable/>);
    const resizableInstance = component.instance() as Resizable;

    expect(resizableInstance.state).toMatchObject({
      isDragging: false,
      dragStart: null,
    });

    component.find('ResizeArea').simulate('mousedown', { clientX: 100 } as any);

    expect(resizableInstance.state).toMatchObject({
      isDragging: true,
      dragStart: 100,
    });
  });

  it('should set state on mouse move if drag started', () => {
    const component = shallow(<Resizable/>);
    const resizableInstance = component.instance() as Resizable;
    resizableInstance.setState({
      isDragging: true,
      dragStart: 50,
      width: 100,
    });

    document.dispatchEvent(mockEvent('mousemove', { clientX: 100 }));

    expect(resizableInstance.state).toMatchObject({
      isDragging: true,
      dragStart: 100,
      width: 50,
    });
  });

  it('should not set state on mouse move if drag not started', () => {
    const component = shallow(<Resizable/>);
    const resizableInstance = component.instance() as Resizable;
    resizableInstance.setState({
      isDragging: false,
      dragStart: null,
      width: 100,
    });

    document.dispatchEvent(mockEvent('mousemove', { clientX: 100 }));

    expect(resizableInstance.state).toMatchObject({
      isDragging: false,
      dragStart: null,
      width: 100,
    });
  });

  it('should set state on drag end', () => {
    const component = shallow(<Resizable/>);
    const resizableInstance = component.instance() as Resizable;
    resizableInstance.setState({
      isDragging: true,
      dragStart: 0,
    });

    document.dispatchEvent(new MouseEvent('mouseup'));

    expect(resizableInstance.state).toMatchObject({
      isDragging: false,
      dragStart: null,
    });
  });
});
