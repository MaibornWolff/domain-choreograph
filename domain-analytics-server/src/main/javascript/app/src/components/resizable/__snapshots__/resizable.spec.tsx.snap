// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resizable should render 1`] = `
<styled.div
  style={
    Object {
      "width": 350,
    }
  }
>
  <styled.div
    onMouseDown={[Function]}
  />
  <styled.div />
</styled.div>
`;
